Documentation update: How To Guides overview (#367)
aragon/osx: packages/contracts/docs/osx/02-how-to-guides/02-plugin-development/index.md

  constructor(IDAO _dao) Plugin(_dao) {}

  function greet() external pure returns (string memory) {

>     return 'Hello world!';

Reviewer: Shouldn't this plugin look more like the storage plugin above?

Author: already updated!

const hre = require('hardhat');

async function main() {
  const [deployer] = await hre.ethers.getSigners();

  console.log('Deploying contracts with the account:', deployer.address);
  console.log('Account balance:', (await deployer.getBalance()).toString());

  const getGreeterSetup = await hre.ethers.getContractFactory('GreeterSetup');
  const GreeterSetup = await getGreeterSetup.deploy();

  console.log('GreeterSetup address:', GreeterSetup.address);
}

main()
  .then(() => process.exit(0))
  .catch(error => {
    console.error(error);
    process.exit(1);
  });
```

On the terminal, we should then see something like this:

```bash
Deploying contracts with the account: 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
Account balance: 10000000000000000000000
GreeterSetup address: 0x5FbDB2315678afecb367f032d93F642f64180aa3
```

### 5. Publish the Plugin to the Aragon protocol

Lastly, we can call the [`createPluginRepoWithFirstVersion` function from Aragon's `PluginRepoFactory`](../../03-reference-guide/framework/plugin/repo/PluginRepoFactory.md) passing it the address of your deployed `GreeterSetup` contract and the first version of your Plugin will be published into the protocol!

> We can do this directly off of calling the function on Etherscan ([make sure to get the right scan and contract address based on your network](https://github.com/aragon/osx/blob/develop/active_contracts.json)) or through locally calling on the method from your project using Ethers.

Reviewer 2: ```suggestion
We can do this directly by calling the function on Etherscan ([make sure to get the right scan and contract address based on your network](https://github.com/aragon/osx/blob/develop/active_contracts.json)) or through locally calling on the method from your project using Ethers.
```

![Etherscan](https://res.cloudinary.com/dacofvu8m/image/upload/v1682466427/Screen_Shot_2023-04-25_at_19.46.58_nlo9p1.png)

If you want to review how to publish your plugin in more depth, review our [How to Publish a Plugin in Aragon OSx guide here](./07-publication//index.md)

### Next Steps

> Congratulations 🎉! You've developed a plugin that every Aragon DAO will be able to use.

Reviewer: ```suggestion
Congratulations 🎉! You have just built a plugin that every Aragon DAO will be able to use.
```
